File: Backend/src/main/java/com/lendingApp/main/dto/UserDetailsResponseDto.java
package com.lendingApp.main.dto;

import java.time.LocalDate;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDetailsResponseDto {
	private UUID userId;

    private String firstName;
    private String middleName;
    private String lastName;
    private String mobile;
    private String email;
    private Boolean isActive;
    private String gender;
    private LocalDate createdAt;
    
    private ProfilePictureResponseDto profilePictureResponseDto;
}

File: FrontEnd/aurofin-frontend/src/app/entity/UserDetailsResponseDto.ts
import { ProfilePictureResponseDto } from './ProfilePictureResponseDto';

export interface UserDetailsResponseDto {
  userId: string; // UUID
  firstName: string;
  middleName?: string;
  lastName: string;
  mobile: string;
  email: string;
  isActive: boolean;
  gender: string;
  createdAt: string; // ISO date string
  profilePictureResponseDto?: ProfilePictureResponseDto | null;
}
